agent-inject/agent/agent.go

```diff
@@ -309,10 +317,27 @@ func (a *Agent) Patch() ([]byte, error) {
 		if err != nil {
 			return patches, err
 		}
-		a.Patches = append(a.Patches, addContainers(
-			a.Pod.Spec.InitContainers,
-			[]corev1.Container{container},
-			"/spec/initContainers")...)
+
+		// Init Containers run sequentially in Kubernetes and sometimes the order in
+		// which they run matters.  This reorders the init containers to put the agent first.
```

Comment:
We may want to provide a brief sentence discussing the context for which a user would want to enable this.

Reply:
Done!